src/helpers/xml-builder.js

```diff
@@ -857,29 +857,37 @@ const computeImageDimensions = (vNode, attributes) => {
   let modifiedWidth;
 
   if (vNode.properties && vNode.properties.style) {
-    if (vNode.properties.style.width) {
-      if (vNode.properties.style.width !== 'auto') {
-        if (pixelRegex.test(vNode.properties.style.width)) {
-          modifiedWidth = pixelToEMU(vNode.properties.style.width.match(pixelRegex)[1]);
-        } else if (percentageRegex.test(vNode.properties.style.width)) {
-          const percentageValue = vNode.properties.style.width.match(percentageRegex)[1];
+    const styleWidth =
+      vNode.properties.style.width > vNode.properties.style['max-width']
+        ? vNode.properties.style['max-width']
+        : vNode.properties.style.width;
```

Comment:
this doesnt sit right with me, considering it could have any type of values like say 100% or auto or 100px or whatever, this comparison should be done post processing the width and max-width values to absolute values like EMUs or px or pt or somehting

Reply:
you are right, my bad i didn't consider them at that point, i have updated the code to do the comparison after calculating the absolute values of them. please check again.